Optimize MailChimp JS block

andrey-legayev opened the pull request:
Avoid initialization of all classes via Magento DI if JS URL has been already cached.
It saves 3-7ms on every shopping cart page load.

gonzaloebiz reviewed (changes requested):
Don't use ObjectManager directly
Don't use the name of a class as a string, use the ::class notation

andrey-legayev committed:
Optimize MailChimp JS block

Avoid initialization of all classes via Magento DI if JS URL has been already cached.

andrey-legayev commented:
I've updated pull request: replaced ObjectManager usage by Proxy configuration in di.xml

gonzaloebiz commented on etc/di.xml:
```diff
@@ -11,6 +11,7 @@
  */
 -->
 <config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
+    <preference for="Ebizmarts\MailChimp\Helper\Data" type="Ebizmarts\MailChimp\Helper\Data\Proxy" />
```
please change this definition for 
`    <type name="Ebizmarts\MailChimp\Block\Mailchimpjs">
        <arguments>
            <argument name="helper" xsi:type="object">Ebizmarts\MailChimp\Helper\Data\Proxy</argument>
        </arguments>
    </type>
`

this way the change affect only to the Mailchimpjs block

andrey-legayev replied:
I did it globally, because I saw usage of the helper in other MailChimp blocks. I think it could also impact performance there (for example on Success page).
But I did not measure performance impact and I didn't get into details of blocks logic.
Let's keep it as you suggested.

andrey-legayev committed:
Limit changes to MailchimpJs block only

gonzaloebiz commented on Block/Mailchimpjs.php:
```diff
@@ -44,6 +46,17 @@ public function __construct(
-        return $this->_helper->getJsUrl($storeId);
+
+        $url    = $this->_scopeConfig->getValue(\Ebizmarts\MailChimp\Helper\Data::XML_MAILCHIMP_JS_URL, ScopeInterface::SCOPE_STORES, $storeId);
+        $active = $this->_scopeConfig->getValue(\Ebizmarts\MailChimp\Helper\Data::XML_PATH_ACTIVE, ScopeInterface::SCOPE_STORES, $storeId);
```
these 2 lines exceeds maximum limit of 120 characters

andrey-legayev replied:
anything else?

andrey-legayev committed:
Reformat code

andrey-legayev committed:
Reformat code

gonzaloebiz reviewed (approved):
Thanks @andrey-legayev
